node/web/api/checkoutApiController.js

```diff
@@ -0,0 +1,215 @@
+        }
+        else {
+          {
+            email = req.cookies.email;
+          }
+        }
+        console.log(email);
+        usersRepo.get(email, function (err, user) {
+            if (err) {
+                exceptions.throwException(err);
+                res.status(400);
+                return;
+            }
+
+            console.log(user.orders);
+            //var orderHistory = new OrderHistory(user.orders);
+            console.log("Oh!!!/api/orderHistory query db succeed!\n");
+
+            res.send(user.orders);
+        });
+    });
+
+
+    // get order
+    router.get('/api/checkout', function (req, res) {
+        console.log("Oh!!!/api/checkout is called with req:\n");
+        //console.log(req);
+
+        // the query should return one order that's supposed to be not completed
+        // but return all the orders for simplicity now?
+        var email = "";
+        if(!req.cookies.email)
+        {
+          email = "Guest";
+        }
+        else {
+          {
+            email = req.cookies.email;
+          }
+        }
+        repo.find({ query: { email: email }}, function (err, orders) {
+            if (err) {
+                exceptions.throwException(err);
+                res.status(400);
+                return;
+            }
+
+            console.log("Oh!!!/api/checkout query db succeed!\n");
+
+            res.send(orders);
+        });
+    });
+
+
+    // save order
+
+    router.post('/api/saveOrder', function (req, res) {
+
+        console.log("a post to /api/saveOrder !!");
+        var email = "";
+        if(!req.cookies.email)
+        {
+          email = "Guest";
+        }
+        else {
+          {
+            email = req.cookies.email;
+          }
+        }
+        repo.update(email, req.body, function (err, result) {
+            if (!err) {
+                res.send("Success");
+            } else {
+                res.send("Fail");
+            }
+        });
+
+    });
+
+    // submit order
+    router.post('/api/submitOrder', function (req, res) {
+
+        console.log("a post to /api/submitOrder !!");
+        var email = "";
+        // currently only save the order, but
+        // consider set a flag in the order table to indicate the order has
+        // been submitted but the payment is pending
+        if(!req.cookies.email)
+        {
+          email = "Guest";
+        }
+        else {
+          {
+            email = req.cookies.email;
+          }
+        }
+        repo.update(email, req.body, function (err, result) {
+            if (!err) {
+                res.send("Success");
+            } else {
+                res.send("Fail");
+            }
+        });
+
+    });
+
+    // payment
+    // temporarily put it here
+    router.post('/payment', function(req, res) {
+
+        console.log("a post to /api/payment !!");
+
+        var stripeToken = req.body.id;
+        // unit: cent
+        var amount = req.body.amount * 100;
```

Comment:
You are trusting the client to tell you how much they are going to pay. This is very hackable (a savvy user could change the price of their books). It would be better to get all of the books they are purchasing and calculate the price on the server. Note that we would want to retrieve the books - not the user's cart, as the prices may have changed since they put the items in their cart.